acceptancetests/assess_network_spaces.py

```diff
@@ -149,18 +148,24 @@ def assert_machine_connectivity(self, client):
         machines = yaml.safe_load(
             client.get_juju_output(
                 'list-machines', '--format=yaml'))['machines']
+        spaces = non_infan_subnets(
+            yaml.safe_load(
+                client.get_juju_output(
+                    'list-spaces', '--format=yaml')))
         # try 0 to 1
-        log.info('Testing ping from Machine 0 Machine 1 (same space)')
-        if not machine_can_ping_ip(
-                client, '0',
-                machines['1']['network-interfaces']['eth0']
-                        ['ip-addresses'][0]):
+        log.info('Testing ping from Machine 0 to Machine 1 (same space)')
+        subnet = spaces['spaces']['space1'].keys()[0]
+        ip_to_ping = [ip for ip in machines['1']['ip-addresses']
+                      if ip_in_cidr(ip, subnet)][0]
+        if not machine_can_ping_ip(client, '0', ip_to_ping):
             raise TestFailure('Ping from 0 to 1 Failed.')
         """Restrictions and access control between spaces is not yet enforced
```

Comment:
Why is this in a block comment?

Reply:
The second ping (from 2 to 3) was meant to assert that ping from one space to another could be blocked but access control between spaces is not available yet. So I commented the whole section out with """ ... """.

Reply:
Ok, I think I would avoid leaving the entire code block commented. Instead perhaps leave a note that you would also want to check ping from Machine 2 to Machine 3 (diff spaces) but that restriction is not yet in effect. In fact, it may even make sense to validate the existing behavior. Then if / when it changes for any reason we'll be alerted to it.

Reply:
Good idea. Added more verbose comment to the doc string and uncommented 2-3 test.